src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpFixAllSpanMappingService.cs

```diff
@@ -19,5 +27,33 @@ internal sealed class CSharpFixAllSpanMappingService : AbstractFixAllSpanMapping
         public CSharpFixAllSpanMappingService()
         {
         }
+
+        protected override async Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetFixAllSpansIfWithinGlobalStatementAsync(
+            Document document, TextSpan diagnosticSpan, FixAllScope fixAllScope, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnosticSpan);
+            if (node.GetAncestorOrThis<GlobalStatementSyntax>() is null)
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
+
+            // Compute the fix all span for the global statements to be fixed.
+            // If the file has non-global statements towards the end, they need to be excluded
+            // from the fix all span.
+            var fixAllSpan = root.FullSpan;
+            var firstNonGlobalStatement = root.ChildNodes().FirstOrDefault(n => n is not GlobalStatementSyntax);
+            if (firstNonGlobalStatement is not null)
+            {
+                // Bail out for error case where a non-global statement precedes global statements.
+                if (root.FullSpan.Start == firstNonGlobalStatement.FullSpan.Start)
+                    return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
```

Comment:
i don't think this check is right.  for example, you could have 

```
using X;

class C { }

if (x) { ... }
```

What i think you simply want to check is if there is a non-global statement that precedes a global statement.

Reply:
Done. thanks